src/task-actions.ts

```diff
@@ -74,6 +80,76 @@ function getFileName(filePath: string) {
     return path.basename(filePath, path.extname(filePath));
 }
 
+export async function starknetCompileCairo1Action(
+    args: TaskArguments,
+    hre: HardhatRuntimeEnvironment
+) {
+    const venvPath = hre.config.starknet.venv;
+    const manifestPath = hre.config.starknet.manifestPath || args?.manifestPath;
+    if (venvPath && !manifestPath) {
+        const msg =
+            "Could not find manifest-path\n" +
+            "The argument '—-manifest-path path/to/Cargo.toml' or manifestPath on hardhat.config.ts should be set.";
+        throw new StarknetPluginError(msg);
+    }
+
+    if (venvPath && path.basename(manifestPath) !== CARGO_FILE && venvPath) {
+        const msg = "The manifest-path must be a path to a Cargo.toml file";
+        throw new StarknetPluginError(msg);
+    }
+
+    const root = hre.config.paths.root;
+    const rootRegex = new RegExp("^" + root);
+
+    const defaultSourcesPath = hre.config.paths.starknetSources;
+    const sourcesPaths: string[] = args.paths || [defaultSourcesPath];
+    const artifactsPath = hre.config.paths.starknetArtifacts;
+
+    let statusCode = 0;
+    for (let sourcesPath of sourcesPaths) {
+        sourcesPath = adaptPath(root, sourcesPath);
+        checkSourceExists(sourcesPath);
+        const files = await traverseFiles(sourcesPath, "*.cairo");
+        const recompiler = new Recompiler(hre);
+        for (const file of files) {
+            console.log("Compiling", file);
+            const suffix = file.replace(rootRegex, "");
+            const fileName = getFileName(suffix);
+            const dirPath = path.join(artifactsPath, suffix);
+            const outputPath = path.join(dirPath, `${fileName}.json`);
+            const casmOutput = path.join(dirPath, `${fileName}${CAIRO_ASSEMBLY_SUFFIX}`);
+            const abiOutput = path.join(dirPath, `${fileName}${ABI_SUFFIX}`);
+
+            fs.mkdirSync(dirPath, { recursive: true });
+            initializeFile(outputPath);
+            initializeFile(casmOutput);
+            initializeFile(abiOutput);
+
+            const executed = await hre.starknetWrapper.cairo1Compile({
+                file,
+                output: outputPath,
+                cairoPath: "",
+                casmOutput,
+                manifestPath,
+                accountContract: args.accountContract,
```

Comment:
Are these accountContract and hintValidation options applicable to cairo1 compilation?